pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -115,23 +124,35 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 	if service.Annotations == nil {
 		service.Annotations = map[string]string{}
 	}
-	if infraConfig.Status.Platform == configv1.AWSPlatformType {
-		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
-		// Set the load balancer for AWS to be as aggressive as Azure (2 fail @ 5s interval, 2 healthy)
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval"] = "5"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout"] = "4"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold"] = "2"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold"] = "2"
-	}
-	// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
-	// GCP load balancers are not customizable and are set to (3 fail @ 8s interval, 1 healthy)
-
-	if infraConfig.Status.Platform == configv1.IBMCloudPlatformType {
-		service.Annotations[iksLBScopeAnnotation] = "public"
+	if infraConfig.Status.PlatformStatus != nil {
+		switch infraConfig.Status.PlatformStatus.Type {
+		case configv1.AWSPlatformType:
+			if ci.Status.EndpointPublishingStrategy.LoadBalancer == nil ||
```

Comment:
This if statement is identical to the if statement starting on line 315 of `deployment.go`. Perhaps it would be beneficial to turn the check into a function to improve code re-use and readability.

Reply:
Commit `72da19a` moves this logic and the logic in `deployment.go` into a separate function.